src/Simple.OData.Client.Core/Simple.OData.Client.Core.csproj

```diff
@@ -6,6 +6,7 @@
     <OutputPath>$(SolutionDir)/build/$(Configuration)/$(AssemblyName)/$(TargetFramework)/</OutputPath>
     <OutDir>$(OutputPath)</OutDir>
     <Decription></Decription>
+    <LangVersion>default</LangVersion>
```

Comment:
I'm actually surprised you could use pattern matching switches without this before!

Reply:
I'm not sure but is pattern matching introduced in C# 7.0?

Reply:
Looks like you're correct for the switch statement:
https://github.com/dotnet/csharplang/blob/main/proposals/csharp-7.0/pattern-matching.md

Looks like it was 8.0 that added the expression switches.